Libs/DICOM/Core/ctkDICOMDisplayedFieldGeneratorRuleFactory.h

```diff
@@ -0,0 +1,86 @@
+/*==============================================================================
+
+  Library:   CTK
+
+  Copyright (c) Pixel Medical Inc. 2021
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0.txt
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  ==============================================================================*/
+
+#ifndef __ctkDICOMDisplayedFieldGeneratorRuleFactory_h_
+#define __ctkDICOMDisplayedFieldGeneratorRuleFactory_h_
+
+#include "ctkDICOMCoreExport.h" 
+
+// Qt includes
+#include <QObject>
+#include <QList>
+
+class ctkDICOMDatabase;
+class ctkDICOMDisplayedFieldGeneratorAbstractRule;
+class ctkDICOMDisplayedFieldGeneratorRuleFactoryCleanup;
+
+/// \ingroup SlicerRt_QtModules_Segmentations
+/// \class ctkDICOMDisplayedFieldGeneratorRuleFactory
+/// \brief Singleton class managing displayed field generator rules
+class CTK_DICOM_CORE_EXPORT ctkDICOMDisplayedFieldGeneratorRuleFactory : public QObject
+{
+  Q_OBJECT
+
+public:
+  /// Instance getter for the singleton class
+  /// \return Instance object
+  Q_INVOKABLE static ctkDICOMDisplayedFieldGeneratorRuleFactory* instance();
+
+  /// Allows cleanup of the singleton at application exit
+  static void setInstance(ctkDICOMDisplayedFieldGeneratorRuleFactory* instance);

```

Comment:
I don't see any specific problems with this, but I think it is unusual to expose this on a public interface. Could we just add a cleanup function that would remove all the rules (if removing them in the destructor is problematic)?